From 46e661b1603a9b79c49abd5998af5d9a14585f97 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Thu, 24 Apr 2025 17:52:24 +0200
Subject: [PATCH 1693/1695] dts: stm32: change scmi for CM33 shared memory
 address on stm32mp21 soc

Use a dedicated memory region for the shared memory used for SCMI commands
between CA35_S an CM33_NS.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5ad753263a265df6f90c5235c6a72f50bca77fd4 dts: stm32: change scmi for CM33 shared memory address on stm32mp21 soc]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: Ia1d4a47a5251e107f52c6a35e729603398977998
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/453946
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi b/core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi
index 6985f1d4f..6db560b77 100644
--- a/core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi
+++ b/core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi
@@ -858,7 +858,7 @@
 			scmi-channel-id = <0>;
 			mbox-names = "scmi_cm33";
 			mboxes = <&ipcc1 RIF_IPCC_CPU1_CHANNEL(16)>;
-			shmem = <&ipc_shmem>;
+			shmem = <&scmi_cid2_ns>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-- 
2.47.2

